vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java

```diff
@@ -759,30 +783,37 @@ private int decodeWayNodes(boolean doubleDelta, MapElement e, int length, boolea
                 deltaLat = buffer[pos];
                 deltaLon = buffer[pos + 1];
             }
-            lat += deltaLat;
-            lon += deltaLon;
+            rawLat += deltaLat;
+            rawLon += deltaLon;
+
+            lat = mTileProjection.projectLat(rawLat);
+            lon = mTileProjection.projectLon(rawLon);
 
             if (pos == length - 2) {
                 boolean line = isLine || (lon != firstLon || lat != firstLat);
 
-                if (line) {
-                    outBuffer[outPos++] = lon;
-                    outBuffer[outPos++] = lat;
-                    cnt += 2;
-                }
-
                 if (e.type == GeometryType.NONE)
                     e.type = line ? LINE : POLY;
 
-            } else /*if ((deltaLon > minDeltaLon || deltaLon < -minDeltaLon
-                    || deltaLat > minDeltaLat || deltaLat < -minDeltaLat)
-                    || e.tags.contains("natural", "nosea"))*/ {
-                // Avoid additional simplification
-                // https://github.com/mapsforge/vtm/issues/39
-                outBuffer[outPos++] = lon;
-                outBuffer[outPos++] = lat;
-                cnt += 2;
+                if (!line) {
```

Comment:
Would be nice to keep all code structure and change or introduce inside the improvements without change current structure, so can see only pull request own changes easily in future.

Reply:
> Why change label position code which defines label points?
> Can work only with lines and avoid other (point) geometries.

Because of the mapfile format:
A Labelposition is stored in delta-lat and delta-long from first wayNode (way coordinate)
So there is need to set the labelposition while reading in the way.

When doing it later we only have access to the projected point, which is then useless when we want to add the delta-lat and delta-long.

Reply:
please see if this is now resloved. :)